daft/table/table_io.py

```diff
@@ -235,6 +239,9 @@ def read_csv(
             f,
             parse_options=pacsv.ParseOptions(
                 delimiter=csv_options.delimiter,
+                quote_char=csv_options.quote,
+                escape_char=csv_options.escape_char,
+                invalid_row_handler=skip_comment(csv_options.comment),
```

Comment:
Ah @suriya-ganesh, the `invalid_row_handler` config for pyarrow's `ParseOptions` wasn't added until pyarrow 7.0.0, and the failing CI jobs are running compatibility testing with pyarrow 6.0.1! https://arrow.apache.org/docs/6.0/python/generated/pyarrow.csv.ParseOptions.html

I think that it would be fair to raise an error if `csv_options.comment is not None` and we're using pyarrow < 7.0.0, e.g.
```python
from daft.utils import ARROW_VERSION

if csv_options.comment is not None and ARROW_VERSION < (7, 0, 0):
    raise ValueError("pyarrow < 7.0.0 doesn't support handling comments in CSVs, please upgrade pyarrow to 7.0.0+.")
``` 

Another thing: we should be sure to pass `invalid_row_handler=None` if `csv_options.comment=None` (the default), to preserve the default pyarrow behavior. It currently looks like the default `None` would be passed to `skip_comment`, which would then fail with a `TypeError` if the returned invalid row handler was ever used. We could change `skip_comment` to:
```python
def skip_comment(comment: str | None) -> Callable | None:
    if comment is None:
        return None
    else:
        return lambda row: 'skip' if row.text.startswith(comment) else 'error'
```